pallets/cash/src/internal/miner.rs

```diff
@@ -52,24 +55,16 @@ impl std::ops::Deref for InherentDataProvider {
 
 #[cfg(feature = "std")]
 #[async_trait::async_trait]
-impl sp_inherents::InherentDataProvider for InherentDataProvider {
+impl<T: Config + std::marker::Sync + std::marker::Send> sp_inherents::InherentDataProvider
+    for InherentDataProvider<T>
+{
     fn provide_inherent_data(
         &self,
         inherent_data: &mut InherentData,
     ) -> Result<(), sp_inherents::Error> {
-        let miner_address_str = runtime_interfaces::validator_config_interface::get_miner_address()
-            .ok_or(sp_inherents::Error::Application(Box::from(
-                "no miner address",
-            )))?;
-
-        let miner_address = our_std::str::from_utf8(&miner_address_str).map_err(|_| {
-            sp_inherents::Error::Application(Box::from("invalid miner address bytes"))
-        })?;
-
-        let chain_account: ChainAccount = our_std::str::FromStr::from_str(miner_address)
-            .map_err(|_| sp_inherents::Error::Application(Box::from("invalid miner address")))?;
-
-        inherent_data.put_data(INHERENT_IDENTIFIER, &chain_account)
+        let validator = core::get_current_validator::<T>()
```

Comment:
this is a function accessing runtime storage I assume?
I think you'll need to introduce a runtime api that allows you to access that data (and that doesn't depend on a `<T: Config>`) similar to e.g. the `GrandpaApi` here: https://github.com/paritytech/substrate/blob/4e75f511e1357cf54c31d9e82d0cf3e39a2b8c28/bin/node/runtime/src/lib.rs#L1307